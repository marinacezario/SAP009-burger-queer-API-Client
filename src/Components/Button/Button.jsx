import React from 'react';

export const Button = ({ id, text, type, alt, value, className }) => {
  return (
    <button
      id={id}
      type={type}
      aria-label={alt} 
      value={value}
      className={className}
    >
      {text}
    </button>
  );
};
