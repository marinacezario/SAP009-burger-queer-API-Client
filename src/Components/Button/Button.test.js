import React from "react";
import { render, screen } from "@testing-library/react";
import { Button } from "./Button";
import "@testing-library/jest-dom";

describe("Button component", () => {
  test("renders without crashing", () => {
    render(
      <Button id="submit-button" type="submit" value="login">
        login
      </Button>
    );
    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(button).toHaveAttribute("type", "submit");
    expect(button).toHaveAttribute("value", "login");
  });
});
