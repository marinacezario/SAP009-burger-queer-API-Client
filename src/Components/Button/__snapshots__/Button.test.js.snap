// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders without crashing 1`] = `
<button
  id="submit-button"
  type="submit"
  value="login"
>
  login
</button>
`;
